src/components/directory-content.jsx

```diff
@@ -487,360 +474,44 @@ const DirectoryContent = () => {
-            setChildrenMetadata({});
-            setIsMissingDataAfterDirChange(false);
-            return;
-        }
-
-        let metadata = {};
-        let childrenToFetchElementsInfos = Object.values(currentChildren)
-            .filter((e) => !e.uploading)
-            .map((e) => e.elementUuid);
-        if (childrenToFetchElementsInfos.length > 0) {
-            fetchElementsInfos(childrenToFetchElementsInfos)
-                .then((res) => {
-                    res.forEach((e) => {
-                        metadata[e.elementUuid] = {
-                            name: e.elementName,
-                            subtype: e.specificMetadata
-                                ? e.specificMetadata.type
-                                : null,
-                            format: e.specificMetadata?.format ?? null,
-                            specificMetadata: e.specificMetadata,
-                        };
-                    });
-                })
-                .catch((error) => {
-                    if (Object.keys(currentChildrenRef.current).length === 0) {
-                        handleError(error.message);
-                    }
-                })
-                .finally(() => {
-                    // discarding request for older directory
-                    if (currentChildrenRef.current === currentChildren) {
-                        setChildrenMetadata(metadata);
-                        setIsMissingDataAfterDirChange(false);
-                    }
-                });
-        }
-        setSelectedUuids(new Set());
-    }, [handleError, currentChildren, currentChildrenRef]);
+    const isActiveElementUnchecked = useMemo(
+        () =>
+            activeElement &&
+            !checkedRows.find(
+                (children) => children.elementUuid === activeElement.elementUuid
+            ),
+        [activeElement, checkedRows]
+    );
 
-    const getSelectedChildren = () => {
-        let selectedChildren = [];
-        if (currentChildren?.length > 0) {
-            // Adds the previously selected elements
-            if (selectedUuids?.size) {
-                selectedChildren = currentChildren
-                    .filter(
-                        (child) =>
-                            selectedUuids.has(child.elementUuid) &&
-                            child.elementUuid !== activeElement?.elementUuid
-                    )
-                    .map((child) => {
-                        return {
-                            subtype:
-                                childrenMetadata[child.elementUuid]?.subtype,
-                            hasMetadata:
-                                childrenMetadata[child.elementUuid] !==
-                                undefined,
-                            ...child,
-                        };
-                    });
-            }
+    const handleRowSelected = useCallback(() => {
+        setCheckedRows(computeCheckedElements(gridRef, childrenMetadata));
```

Comment:
why do we need to add children metadata here ?
it is because we need them to display tooltips or not ?

Reply:
They are needed for the content toolbar as well as the contextual menu